node-zerox/src/utils.ts

```diff
@@ -230,15 +247,65 @@ export const convertPdfToImages = async ({
       responseType: "buffer",
     });
     await Promise.all(
-      convertResults.map(async (result) => {
+      convertResults.map(async (result, idx) => {
         if (!result || !result.buffer) {
           throw new Error("Could not convert page to image buffer");
         }
         if (!result.page) throw new Error("Could not identify page data");
         const paddedPageNumber = result.page.toString().padStart(5, "0");
 
-        // Correct the image orientation
-        const correctedBuffer = await correctImageOrientation(result.buffer);
+        const image = sharp(result.buffer);
+
+        if (correctOrientation) {
+          const suggestion = await suggestRotation(image, idx + 1);
+
+          if (suggestion) {
+            console.log(`Rotating page ${idx + 1} ${suggestion.rotation} degrees`)
+            image.rotate(suggestion.rotation);
+          }
+        }
+
+        if (trimEdges) {
+          image.trim();
+        }
+
+        let { data: correctedBuffer, info } = await image.toBuffer({
+          resolveWithObject: true,
+        });
+
+        if (
+          !correctOrientation &&
+          Math.abs(info.trimOffsetTop || 0) > Y_WHITESPACE_THRESHOLD
+        ) {
+          try {
+            // Increase the image size and try again.
+            // We want to maximize the content
+            const newOptions = {
+              ...options,
+              height: calculateNewHeight(options.height, info.height),
+            };
+
+            const convert = fromPath(localPath, newOptions);
+
+            const result = await convert(idx + 1, {
+              responseType: "buffer",
+            });
+
+            const newTrimmedImage = sharp(result.buffer).trim();
+
+            // Uncomment to see how the image looks
+            // await newTrimmedImage.toFile(`output-${idx}.png`)
```

Comment:
Remove comment. This should be handled by passing in `cleanup: false`. You could just make that conditional.